lib/EmulationManager.js

```diff
@@ -75,4 +75,14 @@ class EmulationManager {
   }
 }
 
+/**
+ * @typedef {Object} EmulationManager.Viewport
```

Comment:
Oh man I love what you've done here.

I think I need to get back to my PR in typescript that allows generating .d.ts files from js files when checkJs is enabled.

Would be great to have code completion when writing puppeteer code.